### PR TITLE
moved some more things to keel-api, merged some other modules

### PR DESCRIPTION
**Pull request by robfletcher**
Part of  #769

I merged `keel-actuator`, `keel-plugin`, and `keel-veto` into `keel-core`. Since core is the engine for running Keel it didn't make sense to me to have those things separate any more.

I also migrated `TaskLauncher`, `Monikered`, and all the `Locations` stuff to `keel-api` with the eventual goal of having `keel-bakery-plugin`, `keel-ec2-plugin`, and `keel-titus-plugin` not having to depend on `keel-core`.

### CODE REVIEW

File: keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt [hunk 1]
```diff
@@ -30,12 +29,12 @@ interface TaskLauncher {
...
     user: String,
     application: String,
-    notifications: List<OrcaNotification>,
+    notifications: Collection<NotificationConfig>,
     subject: String,
     description: String,
     correlationId: String,
     stages: List<Map<String, Any?>>,
-    artifacts: List<Artifact> = emptyList()
+    artifacts: List<Map<String, Any?>> = emptyList()
```
Comment (robfletcher) [inline keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt]: I don't want to have `keel-api` depend on `kork-artifacts`

File: keel-api-jackson/src/main/kotlin/com/netflix/spinnaker/keel/json/mixins/LocatableMixin.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.netflix.spinnaker.keel.json.mixins
+
+import com.fasterxml.jackson.annotation.JacksonInject
+import com.netflix.spinnaker.keel.api.Locations
+
+interface LocatableMixin<T : Locations<*>> {
```
Comment (luispollo) [inline keel-api-jackson/src/main/kotlin/com/netflix/spinnaker/keel/json/mixins/LocatableMixin.kt]: Side-comment: I understand the reason, but having to do all this in code as opposed to using annotations feels like a step backwards... 🙁 What's the issue with annotations and plugins?

File: keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Monikered.kt [hunk 3]
```diff
@@ -17,14 +17,15 @@
  */
 package com.netflix.spinnaker.keel.api
 
-import com.fasterxml.jackson.annotation.JsonIgnore
-
 /**
- * An object which has a moniker
+ * An resource that has a moniker-compliant name.
```
Comment (luispollo) [inline keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Monikered.kt]:
```suggestion
 * A resource that has a moniker-compliant name.
```

File: keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt [hunk 4]
```diff
@@ -30,12 +29,12 @@ interface TaskLauncher {
   suspend fun submitJob(
     user: String,
     application: String,
-    notifications: List<OrcaNotification>,
+    notifications: Collection<NotificationConfig>,
```
Comment (luispollo) [inline keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt]: Shouldn't this be a `Set` instead?

Comment (robfletcher) [reply keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt]: Do you mean because we want to guarantee uniqueness of elements?

Comment (luispollo) [reply keel-api/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt]: Yes. I would expect we wouldn't allow duplicate/redundant notification configs, in this case.

**Pull request merged**